lib/Dialect/TritonGPU/Transforms/Utility.cpp

```diff
@@ -635,4 +635,180 @@ void populateForOpDeadArgumentElimination(RewritePatternSet &patterns) {
   patterns.add<ForOpDeadArgElimination>(patterns.getContext());
 }
 
+// mfma instruction selection logic
+static MfmaTypeId convertTypesToId(mlir::Type dataTypeA, mlir::Type dataTypeB) {
+  if (dataTypeA.isF32() && dataTypeB.isF32()) {
+    return MfmaTypeId::Fp32TyId;
+  }
+  if (dataTypeA.isF16() && dataTypeB.isF16()) {
+    return MfmaTypeId::Fp16TyId;
+  }
+  if (dataTypeA.isBF16() && dataTypeB.isBF16()) {
+    return MfmaTypeId::Bf16TyId;
+  }
+  if (dataTypeA.isInteger(8) && dataTypeB.isInteger(8)) {
+    return MfmaTypeId::I8TyId;
+  }
+  if (dataTypeA.isFloat8E4M3FNUZ() && dataTypeB.isFloat8E4M3FNUZ()) {
+    return MfmaTypeId::Fp8Fp8TyId;
+  }
+  if (dataTypeA.isFloat8E4M3FNUZ() && dataTypeB.isFloat8E5M2FNUZ()) {
+    return MfmaTypeId::Fp8Bf8TyId;
+  }
+  if (dataTypeA.isFloat8E5M2FNUZ() && dataTypeB.isFloat8E4M3FNUZ()) {
+    return MfmaTypeId::Bf8Fp8TyId;
+  }
+  if (dataTypeA.isFloat8E5M2FNUZ() && dataTypeB.isFloat8E5M2FNUZ()) {
+    return MfmaTypeId::Bf8Bf8TyId;
+  }
+  llvm_unreachable("Unsupported input argument type.");
+}
+
+using MfmaInsnGroupMap = llvm::DenseMap<MfmaInsnGroupSelectKey, MfmaInsnAttr,
+                                        MfmaInsnGroupSelectKeyInfo>;
+
+auto getMfmaInsnGroupAttrMap = []() -> const MfmaInsnGroupMap & {
+  static MfmaInsnGroupMap MfmaInsnMap{
+      // f32
+      // mfma_f32_32x32x2f32
+      {{32, MfmaTypeId::Fp32TyId, 1},
+       {32, 32, 2, 1, ROCDL::mfma_f32_32x32x2f32::getOperationName()}},
+      {{32, MfmaTypeId::Fp32TyId, 2},
+       {32, 32, 2, 1, ROCDL::mfma_f32_32x32x2f32::getOperationName()}},
+      {{32, MfmaTypeId::Fp32TyId, 3},
+       {32, 32, 2, 1, ROCDL::mfma_f32_32x32x2f32::getOperationName()}},
+      // mfma_f32_16x16x4f32
+      {{16, MfmaTypeId::Fp32TyId, 1},
+       {16, 16, 4, 1, ROCDL::mfma_f32_16x16x4f32::getOperationName()}},
+      {{16, MfmaTypeId::Fp32TyId, 2},
+       {16, 16, 4, 1, ROCDL::mfma_f32_16x16x4f32::getOperationName()}},
+      {{16, MfmaTypeId::Fp32TyId, 3},
+       {16, 16, 4, 1, ROCDL::mfma_f32_16x16x4f32::getOperationName()}},
+      // mfma_f32_4x4x1f32
+      {{4, MfmaTypeId::Fp32TyId, 1},
+       {4, 4, 16, 1, ROCDL::mfma_f32_4x4x1f32::getOperationName()}},
+      {{4, MfmaTypeId::Fp32TyId, 2},
+       {4, 4, 16, 1, ROCDL::mfma_f32_4x4x1f32::getOperationName()}},
+      // mfma_f32_4x4x1_16B_f32
```

Comment:
I suppose this line is redundant?
```suggestion
```

Reply:
This is the instruction name on MI300.

Reply:
ok, I just noticed that ROCDL operation name is the same here.